sympy/solvers/diophantine.py

```diff
@@ -1040,13 +1040,13 @@ def _diop_quadratic(var, coeff, t):
 
         if D < 0:
             for solution in solns_pell:
-                s1 = P*Matrix([solution[0], solution[1]]) + Q
-                s2 = P*Matrix([-solution[0], solution[1]]) + Q
-                try:
-                    sol.add(tuple([as_int(_) for _ in s1]))
-                    sol.add(tuple([as_int(_) for _ in s2]))
-                except ValueError:
-                    pass
+                for X_i in [-solution[0], solution[0]]:
```

Comment:
Yes you are right,
```python
In [1]: diophantine(x**2 + y**2 -5)
Matrix([[2], [1]])
Matrix([[-2], [1]])
Matrix([[1], [2]])
Matrix([[-1], [2]])
Out[1]: set([(-2, 1), (-1, 2), (1, 2), (2, 1)])

In [2]: diophantine(x**2 + y**2 -5,permute = True)
Matrix([[2], [1]])
Matrix([[-2], [1]])
Matrix([[1], [2]])
Matrix([[-1], [2]])
Out[2]: set([(-2, -1), (-2, 1), (-1, -2), (-1, 2), (1, -2), (1, 2), (2, -1), (2, 1)])
```
I changed it with the orignal code and I am getting this output. 
But there is one problem too.
Considering the case of `10*x**2 - 6*x + 10*y**2 - 14*y - 52548`
```python
diophantine(10*x**2 - 6*x + 10*y**2 - 14*y - 52548)
diophantine(10*x**2 - 6*x + 10*y**2 - 14*y - 52548,python = True)
```
both are giving same number of solutions, i.e. - 6 but there are 16 solutions for it.
`set([(-72, 6), (-70, -17), (-68, 25), (-61, -38), (-59, -41), (-24, 69), (-5, 73), (11, -71), (18, 71), (33, -64), (39, 62), (42, 60), (49, -53
), (54, -48), (65, -32), (72, -10)])`